instructor_notes/fastq_subset/generate_fastq_subsets.sh

```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
+set -euo pipefail
+
+# This script downloads and subsets FASTQ files which can be used during workshops when
+#  internet speed is limited and prohibits timely downloading of the full FASTQ files.
+
+
+# Function to curl, subset, and zip a FASTQ file
+# Provide a single argument: the FASTQ name, without `.gz` extension
+function subset_fastq() {
+    # Grab the file
+    curl -sO https://sra-download.ncbi.nlm.nih.gov/traces/sra63/SRZ/011518/SRR11518889/$1.gz
+
+    # Unzip and take first 10000 records (40000 lines)
+    gunzip -f $1.gz \
```

Comment:
The reason this works without error now is that `gunzip -f` decompresses the whole file and writes it out, which is why you now have to remove the file as well. Doesn't really matter, but it does lose the advantages of not downloading or decompressing the whole file.

Reply:
> The reason this works without error now is that gunzip -f decompresses the whole file and writes it out, which is why you now have to remove the file as well. 

I added this in _after_ confirming this script worked, since it was always prompting for y/n about removing. I'm not sure what difference you are experiencing.

Reply:
Oh, I was confused here: it wasn't the `-f`, it was the absence of `-c`. So with `-c` we only write to the pipe, not a file, but that gives the broken pipe error. Which it turns out is fine if we leave off `o pipefile` from the set line: `set -eu` is fine. We don't need `-c` for `gunzip` if it is taking input from stdin. 
So for me, the following works with `set -eu`. But again, this is so minor as to not need an update!
```
 curl -s https://sra-download.ncbi.nlm.nih.gov/traces/sra63/SRZ/011518/SRR11518889/$1.gz \
    | gunzip \
    | head -n 40000 \
    | gzip \
    > subset-${1}.gz
```

Reply:
aha cool makes sense!